src/polkadotApi/PolkadotApiWebView.tsx

```diff
@@ -0,0 +1,445 @@
+  const setCryptoUtilState = useSetRecoilState(cryptoUtilState);
+
+  React.useEffect(() => {
+    if (isWebviewLoaded) {
+      setCryptoUtilState({
+        generateMnemonic: (length?: MnemonicLength) =>
+          new Promise((resolve) => {
+            resolversRef.current.resolveMnemonic = resolve;
+            webViewRef.current?.postMessage(generateMnemonic.getAction({length}));
+          }),
+        verifyMnemonic: (mnemonic: string) =>
+          new Promise((resolve) => {
+            resolversRef.current.resolveVerifyMnemonic = resolve;
+            webViewRef.current?.postMessage(validateMnemonic.getAction({mnemonic}));
+          }),
+      });
+    }
+  }, [isWebviewLoaded, setCryptoUtilState, webViewRef, resolversRef]);
+}
+
+function useKeyringUtils(isWebviewLoaded: boolean, webViewRef: WebViewRef, resolversRef: ResolversRef) {
+  const setKeyringState = useSetRecoilState(keyringState);
+
+  React.useEffect(() => {
+    if (isWebviewLoaded) {
+      setKeyringState({
+        createAccount: (mnemonic: string) =>
+          new Promise((resolve) => {
+            resolversRef.current.resolveCreateAccount = resolve;
+            webViewRef.current?.postMessage(createAccount.getAction({mnemonic}));
+          }),
+        addAccount: (payload: AddAccountPayload) =>
+          new Promise((resolve) => {
+            resolversRef.current.resolveAddAccount = resolve;
+            webViewRef.current?.postMessage(addAccount.getAction(payload));
+          }),
+        addExternalAccount: (payload: AddExternalAccountPayload) =>
+          new Promise((resolve) => {
+            resolversRef.current.resolveAddExternalAccount = resolve;
+            webViewRef.current?.postMessage(addExternalAccount.getAction(payload));
+          }),
+        forgetAccount: (address: string) => {
+          webViewRef.current?.postMessage(forgetAccount.getAction({address}));
+        },
+        restoreAccount: (payload: RestoreAccountPayload) =>
+          new Promise((resolve, reject) => {
+            resolversRef.current.restoreAccountPromise.resolve = resolve;
+            resolversRef.current.restoreAccountPromise.reject = reject;
+            webViewRef.current?.postMessage(restoreAccount.getAction(payload));
+          }),
+        exportAccount: (payload: ExportAccountPayload) =>
+          new Promise((resolve, reject) => {
+            resolversRef.current.exportAccountPromise.resolve = resolve;
+            resolversRef.current.exportAccountPromise.reject = reject;
+            webViewRef.current?.postMessage(exportAccount.getAction(payload));
+          }),
+        verifyCredentials: (credentials: SignCredentials) =>
+          new Promise((resolve) => {
+            resolversRef.current.resolveVerifyCredentials = resolve;
+            webViewRef.current?.postMessage(verifyCredentials.getAction(credentials));
+          }),
+        sign: (message: string, credentials: SignCredentials) =>
```

Comment:
Let's add the return type here so we don't have to do type assertion in the `TxContext`

Reply:
```
Promise<{signed: `0x${string}`}>